jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/SessionManager.java

```diff
@@ -196,7 +196,8 @@ public void save() {
 
     protected void save (boolean force)
     {
-      logger.fine("Session " + getId() + "is"+(dirty?" dirty":" not dirty")+(force || dirty ? " saving":" not saving"));
+      if (logger.isLoggable(Level.FINE))
+        logger.fine("Session " + getId() + "is"+(dirty?" dirty":" not dirty")+(force || dirty ? " saving":" not saving"));
```

Comment:
I would consider using formatter for such a string.

I assume `getId()` does not have to end with space and therefore you probably want to change "is" to " is"

spaces around " + ", " ? " and " : " - http://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace